### DISCUSSION TITLE
`black` catches format issues in jupyter notebooks?

### DISCUSSION THREAD

**Discussion by NimaSarajpoor** [TDAmeritrade/stumpy#642] [Help: Other Questions] [answered]
So, I tried to run `black stumpy/.` and `black tests/.`. Both are okay!

However, when I run `./test.sh` (or the custom test `./test.sh custom 1 ...` , the `black` finds some formatting issues in jupyter notebooks and exit with error.  The version of black in my environment is `22.6.0`.

Could someone help me out regarding this matter? (@seanlaw)

**Comment by seanlaw** [comment, marked as answer]
That doesn't make sense. Can you provide your full `./test.sh custom 1 ...` command?

I'm running the same version of `black` and didn't see anything when running `./setup.sh && test.sh`.

**Reply by NimaSarajpoor** [reply]
@seanlaw 
Sure.

```
(stumpy) nimas (TopK_MatrixProfile) stumpy
$ ./setup.sh && ./test.sh
Found existing installation: stumpy 1.11.1
Uninstalling stumpy-1.11.1:
  Would remove:
    c:\users\nimas\anaconda3\envs\stumpy\lib\site-packages\stumpy-1.11.1.dist-info\*
    c:\users\nimas\anaconda3\envs\stumpy\lib\site-packages\stumpy\*
Proceed (Y/n)?   Successfully uninstalled stumpy-1.11.1
Processing e:\+machine_learning_journey\contributions\stumpy
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Requirement already satisfied: numpy>=1.17 in c:\users\nimas\anaconda3\envs\stumpy\lib\site-packages (from stumpy==1.11.1) (1.21.6)
Requirement already satisfied: scipy>=1.5 in c:\users\nimas\anaconda3\envs\stumpy\lib\site-packages (from stumpy==1.11.1) (1.8.0)
Requirement already satisfied: numba>=0.54 in c:\users\nimas\anaconda3\envs\stumpy\lib\site-packages (from stumpy==1.11.1) (0.55.1)
Requirement already satisfied: setuptools in c:\users\nimas\anaconda3\envs\stumpy\lib\site-packages (from numba>=0.54->stumpy==1.11.1) (62.1.0)
Requirement already satisfied: llvmlite<0.39,>=0.38.0rc1 in c:\users\nimas\anaconda3\envs\stumpy\lib\site-packages (from numba>=0.54->stumpy==1.11.1) (0.38.0)
Building wheels for collected packages: stumpy
  Building wheel for stumpy (setup.py): started
  Building wheel for stumpy (setup.py): finished with status 'done'
  Created wheel for stumpy: filename=stumpy-1.11.1-py3-none-any.whl size=145770 sha256=2556a977f41a0a0f376ba0546153d5871a699bc777ab16b900a44852c7654d0b
  Stored in directory: C:\Users\nimas\AppData\Local\Temp\pip-ephem-wheel-cache-430ffphx\wheels\bd\a0\d4\260c4e0536ca332f29f477f910225e6884b8a3b9f79f7cb94b
Successfully built stumpy
Installing collected packages: stumpy
Successfully installed stumpy-1.11.1
Cleaning Up
Checking Black Code Formatting
--- docs\log_analysis\Log_analaysis.ipynb       2022-06-09 01:21:53.220933 +0000:cell_1
+++ docs\log_analysis\Log_analaysis.ipynb       2022-07-10 16:27:22.821746 +0000:cell_1

## a LOT of messages here, so I am skipping them! ##

-    plt.axvline(x=x[0]-x.min()+(m+5)*i + 11, alpha=0.3)
-    plt.axvline(x=x[0]-x.min()+(m+5)*i + 15, alpha=0.3, linestyle='-.')
-    plt.plot(x-x.min()+(m+5)*i, y-y.min(), linewidth=3)
+    plt.axvline(x=x[0] - x.min() + (m + 5) * i + 11, alpha=0.3)
+    plt.axvline(x=x[0] - x.min() + (m + 5) * i + 15, alpha=0.3, linestyle="-.")
+    plt.plot(x - x.min() + (m + 5) * i, y - y.min(), linewidth=3)
 plt.show()
would reformat docs\Tutorial_Time_Series_Chains.ipynb

Oh no! \U0001f4a5 \U0001f494 \U0001f4a5
24 files would be reformatted, 87 files would be left unchanged, 11 files would fail to reformat.
Error: pytest encountered exit code 123
```

**Reply by NimaSarajpoor** [reply]
@seanlaw 
Btw, I was developing a feature in `mlxtend` library and I THINK I might have changed something there when I was running test. However, when I faced the error message provided above in `stumpy` environment, I tried `pip uninstall black` and re-install it again. That did not help! Maybe I did something wrong.

**Reply by seanlaw** [reply]
I think it's because you have `pip install 'black[jupyter]'`, which installed some dependencies that forces `.ipynb` files to get checked by `black`.  In my case, I see:

```
Checking Black Code Formatting
Skipping .ipynb files as Jupyter dependencies are not installed.
You can fix this by running ``pip install black[jupyter]``
All done! ✨ 🍰 ✨
83 files would be left unchanged.
```

When I executed `pip install black[jupyter]` and then tried running `black` again, I see the same errors. I just added a fix to exclude `.ipynb` files in `test.sh` when executing `black`. Please merge the latest commits and see if that solves your problem.

**Reply by NimaSarajpoor** [reply]
> I think it's because you have `pip install 'black[jupyter]'`, which installed some dependencies that forces `.ipynb` files to get checked by `black`

Exactly! I actually did `pip install 'black[jupyter]` a few days ago. I got similar errors. However, when I did `pip uninstall black` and re-install it again, the issue was resolved (or maybe I did something else!). 

**But, this time, I couldn't resolve the  issue! (And, I do not remember doing `pip install 'black[jupyter]`)**  I was wondering if  `pip uninstall black` and re-installing it again would solve the issue. Did you try to uninstall the jupyter-related dependencies after executing `pip install black[jupyter]`? 

---

> Please merge the latest commits and see if that solves your problem.

Thanks! Will do so!

**Reply by NimaSarajpoor** [reply]
> Please merge the latest commits and see if that solves your problem.

It works! Thanks.

**Reply by seanlaw** [reply]
Great!

**Reply by NimaSarajpoor** [reply]
@seanlaw 
Should I accept this myself? Because, I cannot find the button "accept" or something like that in my end. So, please let me know if I should accept (close) this. Thanks :)
